packages/orchestration/src/orchestration-api.ts

```diff
@@ -88,6 +88,24 @@ export type CosmosChainAddress = {
   encoding: 'bech32' | 'ethereum';
 };
 
+/**
+ * Info used to identify blockchains across ecosystems
+ * @see {@link https://chainagnostic.org/CAIPs/caip-2}
+ */
+export interface BaseChainInfo<N extends KnownNamespace = KnownNamespace> {
+  /** CAIP-2 namespace, e.g. 'cosmos', 'eip155' */
+  namespace: N;
+  /** CAIP-2 reference, `e.g. `1`, `agoric-3` */
+  reference: N extends 'eip155' ? `${number}` : string;
```

Comment:
not sure this is worth the trouble but no objection

Reply:
> not sure this is worth the trouble

+ 1. I'll leave for now but we can loosen in the future.